api/controller/assignments.js

```diff
@@ -0,0 +1,850 @@
+                                if (err) {
+                                    return res.status(500).send({
+                                        error: "An error has occurred",
+                                        code: 10
+                                    });
+                                } else {
+                                    if (result && result.result && result.result.n == 1) {
+                                        res.send({
+                                            id: assignmentId
+                                        });
+                                    } else {
+                                        res.status(401).send({
+                                            error: "Inexisting assignment id",
+                                            code: 23
+                                        });
+                                    }
+                                }
+                            }
+                        );
+                    } else {
+                        return res.status(401).send({
+                            error: "Assignment already exists",
+                            code: 24
+                        });
+                    }
+                }
+            });
+    });
+};
+
+//private function for filtering and sorting
+function getOptions(req, count, def_sort) {
+    //prepare options
+    var sort_val = typeof req.query.sort === "string" ? req.query.sort : def_sort;
+    var sort_type = sort_val.indexOf("-") == 0 ? "desc" : "asc";
+    var options = {
+        sort: [[sort_val.substring(1), sort_type]],
+        skip: req.query.offset || 0,
+        total: count,
+        limit: req.query.limit || 10
+    };
+
+    //cast to int
+    options.skip = parseInt(options.skip);
+    options.limit = parseInt(options.limit);
+
+    //return
+    return options;
+}
+
+//form query params
+function formPaginatedUrl(route, params, offset, limit) {
+    //set params
+    params.offset = offset;
+    params.limit = limit;
+    var str = [];
+    for (var p in params)
+        if (p && Object.prototype.hasOwnProperty.call(params, p)) {
+            str.push(p + "=" + params[p]);
+        }
+    return "?" + str.join("&");
+}
+
+//update comment 
+exports.updateComment = function (req, res) {
+    //validate
+    if (!mongo.ObjectID.isValid(req.params.assignmentId)) {
+        res.status(401).send({
+            error: "Invalid assignment id",
+            code: 23
+        });
+        return;
+    }
+    if (!req.params.assignmentId) {
+        res.status(400).send({
+            error: "Assignment id is not defined",
+            code: 22
+        });
+        return;
+    }
+    var assignmentId = req.params.assignmentId;
+    var comment = JSON.parse(req.body.comment);
+    if (!req.query.oid) {
+        return res.status(401).send({
+            error: "Invalid object id",
+            code: 23
+        });
+    }
+    var objectId = req.query.oid;
+
+    try {
```

Comment:
Instead of using Try/Catch, can you keep the coding pattern similar to the rest of the code.